src/config_design.config

```diff
@@ -0,0 +1,126 @@
+%% Cloudant_benchmark config changes design review 
+
+%% ----------------------------------------------------------------------------
+%% CONFIGURATION TEMPLATES: (Fogbugz 62351)
+%% ----------------------------------------------------------------------------
+
+{config_templates, ["basho_bench", "cloudant_http", "http_driver"]}.
+
+%% IMPLEMENTATION
+%% Optional configuration template files to predefine parameters/preset default values
+%% Each record would contain something like (Name,Value,ValidatorFun,Description,?MODULE) 
+%%
+%% Config template files could be nested, 
+%%    would allow us to validate-on-load incoming parameters that match
+%%    warn about unknown config params to catch typos
+%%    used to auto-generate a key-value only config file for local config file use, 
+%%    would help dumping ALL config params+doc to run log, not just those provided by user
+%% 
+%% basho_bench_config:load
+%%    would pre-process terms instead of copying directly to application environmant
+%%    
+%%    if "config_templates" call then loads those config template files, 
+%%        saving parameter records into common ETS table
+%%        presetting parameter value defaults into environment
+%%
+%%    else if any templates have been defined,
+%%        checks term against known parameter records
+%%        if known then checks value with ValidatorFun before saving to environment
+%%        else warns about unknown parameters (potential typos) but accepts
+
+%% ----------------------------------------------------------------------------
+%% PER-OPERATIONS CONFIGURATION: https://github.com/cloudant/perf/issues/23 and 29
+%% ----------------------------------------------------------------------------
+%%
+%% Operations would change to allow optional per-operation keygen and valgen
+
+{operations, [
+   %% basho_bench_worker and _stats expect two values - Name, Count 
+   {get, 1},
+
+   %% Using alternate {op,[list]} format keeps 2-tuples so either format could be used
+   %% QUESTION: Do we need ONLY keygen or the OPTION of BOTH keygen and valgen ? 
+   %%    Would we ever need MORE per-op options? 
+
+   {put, [{count,1},  % Alternate count in more complex format, modify existing lookups
+          {key_generator, {int_to_str, {partitioned_sequential_int, 100000}}},
+          {value_generator, {function,http_couchdb_driver,make_json_doc,[]}}
+	 ]}
+   ]}.
+
+%% IMPLEMENTATION:
+%% basho_bench_worker:init 
+%%    turns per-op config into new Keygen/ValueGen entities
+%%    and stores into worker State as per-op-config list
+%%        per_op_config =[{put, [{keygen, KeyGen}, {valgen, ValGen}, ...]}, ... ]
+%% basho_bench_worker:worker_next_op2 
+%%    checks per_op_config, uses as run call arguments,
+%%       potentially stores onto State as current keygen/valgen
+
```

Comment:
@chewbranca 

It isn't too hard to interpret an operation either way but it gets into the whole "operation" mess in basho which requires extra code in multiple places to read, i.e. basho_bench is supporting {OpTag, Count} and {Label, OpTag, Count}, making filenames from OpTags to which I want to add workertype_ prefixes, etc.  This change would add OpTag being Operation or {Module, Operation} on top of all that. 

This would be a good time to clean up the mess and just normalize the operations table on entry so that we just carry around ONE version of the operations at runtime. We would accept different input formats but then save the operations array in a single fully populated tuple, the way we want it to be. 

i.e. {FullOutputFileLabel, {ModuleOrDriver, Operation}, Count, OptionalConfigList}